lib/tasks/generate_terraform.rake

```diff
@@ -16,13 +16,18 @@ task :generate_terraform do
     FileUtils.rm files
   end
 
-  # Load what we want
-  records = YAML.load(File.read(zonefile))['records']
+  # Load configuration
+  config_file = YAML.load_file(zonefile)
+  records = config_file['records']
+  deployment = config_file['deployment']
 
   # Render all the expected files
   providers.each { |current_provider|
-    tf_vars = REQUIRED_ENV_VARS[current_provider.to_sym][:tf]
-    out = generate_terraform_object(current_provider, records, tf_vars)
+    deploy_vars = deployment[current_provider.to_sym]
+
+    abort("Must set deployment options in configuration file") if deploy_vars.empty?
```

Comment:
This validation should be reworked slightly and moved up as it's testing after the assignment has been made, which could cause an issue if `current_provider.to_sym` doesn't exist.